int8[] board_state

